pgmpy/base/AncestralBase.py

```diff
@@ -528,6 +529,124 @@ def get_reachable_nodes(self, node, u_type=None, v_type=None):
                 queue.extend(self.get_neighbors(current, u_type=u_type, v_type=v_type))
         return reachable
 
+    def to_dagitty(self) -> str:
+        """
+        Convert the MAG to a Dagitty string representation.
+        Returns
+        -------
+        str
+            A string in Dagitty format representing the MAG.
+
+        Examples
+        --------
+        >>> from pgmpy.base import MAG
+        >>> mag = MAG()
+        >>> mag.add_edge("X", "Y", "-", ">")
+        >>> mag.add_edge("Z", "Y", "-", ">")
+        >>> print(mag.to_dagitty())
+        dag {
+        X -> Y
+        Z -> Y
+        }
+
+        >>> mag2 = MAG()
+        >>> mag2.add_edge("A", "B", ">", ">")
+        >>> mag2.add_edge("C", "D", "-", "-")
+        >>> print(mag2.to_dagitty())
+        dag {
+        A <-> B
+        C -- D
+        }
+
+        >>> # MAG with latent variables and roles
+        >>> mag3 = MAG()
+        >>> mag3.add_edge("L", "X", "-", ">")
+        >>> mag3.add_edge("X", "Y", "-", ">")
+        >>> mag3.latents = {"L"}
+        >>> mag3 = mag3.with_role("exposure", "X")
+        >>> mag3 = mag3.with_role("outcome", "Y")
+        >>> print(mag3.to_dagitty())
+        dag {
+        L -> X
+        X -> Y
+        L [latents]
+        Y [outcome]
+        X [exposure]
+        }
+
+        Notes
+        -----
+        - Node names are converted to string representations.
+        - If node names contain spaces or special characters, they will be used as-is.
+        - Users should ensure node names are valid in R/dagitty context if needed.
```

Comment:
I would suggest keeping the first statement in the main method description in line 534. I think the rest of the points are a bit obvious, and we can remove the `Notes` section.